src/AntiCSRF.php

```diff
@@ -548,4 +544,34 @@ protected static function noHTML(string $untrusted): string
     {
         return \htmlentities($untrusted, ENT_QUOTES, 'UTF-8');
     }
+
+    /**
+     * Wrapper to support different sources of remote IP addresses.
+     *
+     * @return string
+     */
+    protected function getRemoteIP(): string
+    {
+        if (isset($this->server["HTTP_CF_CONNECTING_IP"]) && filter_var($this->server["HTTP_CF_CONNECTING_IP"], FILTER_VALIDATE_IP))
+        {
+            return $this->server["HTTP_CF_CONNECTING_IP"];
+        }
+
+        if (isset($this->server["HTTP_X_FORWARDED_FOR"]) && filter_var($this->server["HTTP_X_FORWARDED_FOR"], FILTER_VALIDATE_IP))
+        {
+            return $this->server["HTTP_X_FORWARDED_FOR"];
+        }
```

Comment:
Attackers can lie about `X-Forwarded-For`. This should be guarded by a configuration setting.

Reply:
What would you suggest? A configuration setting as in hmac_ip_allow_x_forwarded_for? Or something more general that would include both this option as well as CloudFlare for example?

Reply:
Yes, a configuration setting would be a good idea. Let people opt into this behavior.